cfme/infrastructure/provider/openstack_infra.py

```diff
@@ -55,31 +41,133 @@ def has_nodes(self):
 
     @classmethod
     def from_config(cls, prov_config, prov_key, appliance=None):
-        credentials_key = prov_config['credentials']
-        credentials = cls.process_credential_yaml_key(credentials_key)
-        credential_dict = {'default': credentials}
+        endpoints = [RHOSEndpoint(**prov_config['endpoints']['default']),
```

Comment:
@psav, fixed. 
1. framework takes only existing endpoints from config now
2. endpoints may not have credentials.